tests/common/mocks/KeyValueCache.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <gmock/gmock.h>
+
+#include <olp/core/cache/KeyValueCache.h>
+
+namespace olp {
+namespace testing {
```

Comment:
that's why it is in olp namespace :-)

Reply:
1. You don't put tests in the same namespace as the classes you are testing.
2. You don't use existing namespaces.

Reply:
`olp::testing::mocks` is not the same as `olp::cache`. Also I haven't seen 
> don't use `olp` namespace for tests

anywhere in code convention. Is there a good reason for that? 

I don't extend existing namespaces anywhere in tests/mocks/wherever

Reply:
You extend the olp namespace.
I think the main reason is the namespace pollution, and if this is not enough it is the mear reason for the namespace itself. Using `olp` means you add the "feature" of mocked classes to the olp scope.

Reply:
ok, I was wrong. I meant to say I don't extend "leaf" namespaces. How I see the alternative:

I put tests into `::test`, because `::testing` is not good enought and is taken by Google. The other day somebody I've never met will use SDK as subproject, write in CMakeLists.txt `add_subdirectory(here_olp_sdk)` and write their own tests and put them into `::test`. Why? Because `::testing` is already taken by Google and nobody else will ever put anything in `::test` namespace, right?

On the other side, `::olp` is our safe bay which we have full control over and we can agree on `::olp::testing` for tests.